### PR TITLE
Improve Txr usability

### PR DESCRIPTION
**Pull request by catostrophe**
- define Aux alias via bounds
- add Txr implicit summoners
- add txr syntax: trans/rawTrans
- improve example

### CODE REVIEW

**Comment by catostrophe** [comment]
@oskin1 @FunFunFine I kindly ask you to `publishLocal` a snapshot and try it in your projects. I suppose the change in Aux definition shouldn't break anything (it is needed for `izumi-reflection`).

File: doobie/src/main/scala/tofu/doobie/transactor/Txr.scala [hunk 1]
```diff
@@ -42,11 +42,16 @@ trait Txr[F[_]] {
 }
 
 object Txr {
```
Comment (FunFunFine) [inline doobie/src/main/scala/tofu/doobie/transactor/Txr.scala:44]: I think is better to have scaladoc here with a bit of motivation. Like what types one should use.

File: doobie/src/main/scala/tofu/syntax/doobie/txr.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import fs2.Stream
+import tofu.doobie.transactor.Txr
+
+private[doobie] trait TxrSyntax {
+  implicit def toTxrOps[DB[_], A](dba: DB[A]): TxrOps[DB, A]                     = new TxrOps(dba)
+  implicit def toTxrStreamOps[DB[_], A](dba: Stream[DB, A]): TxrStreamOps[DB, A] = new TxrStreamOps(dba)
+}
+
+private[doobie] final class TxrOps[DB[_], A](private val dba: DB[A]) extends AnyVal {
+  def trans[F[_]](implicit txr: Txr.Aux[F, DB]): F[A]    = txr.trans(dba)
```
Comment (FunFunFine) [inline doobie/src/main/scala/tofu/syntax/doobie/txr.scala:12]: I am a little afraid that such method name shortening would make it harder to understand code at a first glance.
I guess it won't hurt anyone to name it `transact` & `rawTransact`.

Comment (FunFunFine) [reply doobie/src/main/scala/tofu/syntax/doobie/txr.scala:12]: Same for streams one

File: doobie/src/test/scala/tofu/doobie/example/TofuDoobieExample.scala [hunk 3]
```diff
@@ -104,8 +107,11 @@ trait DeptSql[F[_]] {
 }
 
 object DeptSql {
-  def make[DB[_]: Monad: LiftConnectionIO: Logging: Tracing](elh: EmbeddableLogHandler[DB]): DeptSql[DB] =
-    NonEmptyList.of(new LoggingMid[DB], new TracingMid[DB]).reduce attach elh.embedLift(implicit lh => new Impl)
+  def make[DB[_]: Monad: LiftConnectionIO: Logging: Tracing: EmbeddableLogHandler]: DeptSql[DB] = {
+    val aspects = NonEmptyList.of(new TracingMid[DB], new LoggingMid[DB]).reduce
+    val impl    = EmbeddableLogHandler[DB].embedLift(implicit lh => new Impl)
+    aspects attach impl
```
Comment (FunFunFine) [inline doobie/src/test/scala/tofu/doobie/example/TofuDoobieExample.scala:113]: 👍🏾

**Review by FunFunFine — APPROVED** [review]

**Review by oskin1 — DISMISSED** [review]

File: doobie/src/main/scala/tofu/syntax/doobie/txr.scala (hunk 2 above)
Comment (catostrophe) [reply doobie/src/main/scala/tofu/syntax/doobie/txr.scala:12]: I took the names from `doobie` to reduce entropy. I don't want to go the ZIO way.

File: doobie/src/main/scala/tofu/doobie/transactor/Txr.scala [hunk 4]
```diff
@@ -42,11 +42,16 @@ trait Txr[F[_]] {
...
-  type Aux[F[_], DB0[_]]     = Txr[F] { type DB[x] = DB0[x] }
+  type Aux[F[_], DB0[_]]     = Txr[F] { type DB[x] >: DB0[x] <: DB0[x] }
```
Comment (catostrophe) [inline doobie/src/main/scala/tofu/doobie/transactor/Txr.scala:45]: Everything compiles fine with this change (including the example). The only issue with the previous encoding was related to `distage` runtime. I couldn't write a failing test with `izumi-reflect` only and I don't want to involve the whole `distage` lib into tests. I checked (and already knew for a long time) that this fix worked out.

File: doobie/src/main/scala/tofu/syntax/doobie/txr.scala (hunk 2 above)
Comment (FunFunFine) [reply doobie/src/main/scala/tofu/syntax/doobie/txr.scala:12]: Okay, this is legit too.

**Review by Odomontois — APPROVED** [review]

**Pull request merged**